src/fdcache_fdinfo.cpp

```diff
@@ -433,6 +433,8 @@ bool PseudoFdInfo::ParallelMultipartUpload(const char* path, const mp_part_list_
             delete thargs;
             return false;
         }
+
+        AutoLock auto_lock(&upload_list_lock);
```

Comment:
Why do we even need a thread pool?  Creating a thread is much cheaper than the network RPC.

Reply:
It is lighter than network RPC as you say, but thread pools are different.
Since it only issues instructions(semaphore) to threads that have been started(waiting) in advance, there is no startup overhead.